### PR TITLE
[CR] jsonapi: restore kind= to actions + prune nulls

### PR DESCRIPTION
**Pull request by felliott**
Restore `?kind=folder` to new_folder action, `?kind=file` to upload
action, and remove any null actions from response

[#OSF-4845]

### CODE REVIEW

File: waterbutler/core/metadata.py [hunk 1]
```diff
@@ -52,14 +52,20 @@ def _json_api_links(self, resource):
...
-            'new_folder': entity_url if self.is_folder else None,
+        actions = {
             'move': entity_url,
-            'upload': entity_url,
-            'download': entity_url if self.is_file else None,
+            'upload': entity_url + '?kind=file',
             'delete': entity_url,
         }
 
+        if self.is_folder:
```
Comment (chrisseto) [inline waterbutler/core/metadata.py]: This would better done with inheritance

**Comment by abought** [comment]
Some notes for test/ review, courtesy of @felliott 

This PR will change the data returned by a particular URL. Specifically:

> WB has an undocumented meta query parameter that will return the representation with the action links included.
e.g. http://127.0.0.1:7777/v1/resources/p8a63/providers/s3/?meta=
Templated: http://127.0.0.1:7777/v1/resources/{node_id}/providers/{provider_name}/{path_to_file}?meta=
{node_id} == OSF Project Node ID   {path_to_file} == File ID

File: waterbutler/core/metadata.py [hunk 2]
```diff
@@ -140,6 +143,11 @@ def serialized(self):
             'size': self.size,
         })
 
+    def _json_api_links(self, resource):
+        ret = super()._json_api_links(resource)
+        ret['download'] = self._entity_url(resource)
```
Comment (chrisseto) [inline waterbutler/core/metadata.py]: :horse:

**Comment by abought** [comment]
Output looks ok. Thanks for the help getting this running!

Before:
```json
        "links": {
            "download": "http://127.0.0.1:7777/v1/resources/dc2sx/providers/osfstorage/561e769fc4f5fbf72f0333ab",
            "new_folder": null,
            "delete": "http://127.0.0.1:7777/v1/resources/dc2sx/providers/osfstorage/561e769fc4f5fbf72f0333ab",
            "upload": "http://127.0.0.1:7777/v1/resources/dc2sx/providers/osfstorage/561e769fc4f5fbf72f0333ab",
            "move": "http://127.0.0.1:7777/v1/resources/dc2sx/providers/osfstorage/561e769fc4f5fbf72f0333ab"
        },
```


After:
```json
        "links": {
            "upload": "http://127.0.0.1:7777/v1/resources/dc2sx/providers/osfstorage/561e769fc4f5fbf72f0333ab?kind=file",
            "download": "http://127.0.0.1:7777/v1/resources/dc2sx/providers/osfstorage/561e769fc4f5fbf72f0333ab",
            "move": "http://127.0.0.1:7777/v1/resources/dc2sx/providers/osfstorage/561e769fc4f5fbf72f0333ab",
            "delete": "http://127.0.0.1:7777/v1/resources/dc2sx/providers/osfstorage/561e769fc4f5fbf72f0333ab"
        }
```

**Comment by abought** [comment]
CR2 done; no changes requested.

**Comment by chrisseto** [comment]
LGTM

**Pull request merged**